annotation/tests/test_annotations.py

```diff
@@ -235,3 +278,253 @@ def test_create_manifest_json(
+            statement="TEST", params=("testuser",), orig=Exception("TESTING")
+        )
+    )
+    db.rollback = Mock()
+    with patch(
+        "annotation.annotations.main.get_pages_sha",
+        return_value=pages_sha_latest if is_latest else pages_sha_non_latest,
+    ), patch(
+        "annotation.annotations.main.check_docs_identity",
+        return_value=new_equal_latest,
+    ), patch(
+        "annotation.annotations.main.construct_document_links", return_value=[]
+    ):
+        with pytest.raises(DuplicateAnnotationError):
+            construct_annotated_doc(
+                db,
+                annotated_doc.user,
+                None,
+                annotated_doc.job_id,
+                annotated_doc.file_id,
+                doc,
+                TEST_TENANT,
+                s3_file_path,
+                s3_file_bucket,
+                latest_doc,
+                1,
+                is_latest,
+            )
+            assert db.commit.assert_called_once()
+            assert db.rollback.assert_called_once()
+
+
+def test_upload_json_to_minio():
+    test_json = json.dumps({"test": 1})
+    path_to_object = "path"
+    bucket_name = TEST_TENANT
+
+    with patch(
+        "annotation.annotations.main.bd_storage.get_storage"
+    ) as mock_get_storage:
+        mock_get_storage.return_value.upload_obj = Mock()
+
+        upload_json_to_minio(test_json, path_to_object, bucket_name, None)
+        mock_get_storage.assert_called_once_with(bucket_name)
+        mock_get_storage().upload_obj.assert_called_once_with(
+            target_path=path_to_object, file=ANY
+        )
+
+
+def test_upload_pages_to_minio(moto_s3: boto3.resource):
+    pages = [PageSchema(page_num=1, size={}, objs=[])]
+    pages_sha = {"1": "sha1"}
+    s3_path = "path"
+    with patch(
+        "annotation.annotations.main.upload_json_to_minio"
+    ) as mock_upload_json:
+        upload_pages_to_minio(pages, pages_sha, s3_path, TEST_TENANT, moto_s3)
+        path_to_object = f"{s3_path}/{pages_sha[str(pages[0].page_num)]}.json"
+        mock_upload_json.assert_called_once_with(
+            json.dumps(pages[0].dict()), path_to_object, TEST_TENANT, moto_s3
+        )
+
+def test_update_pages_array():
+    resulting_set = update_pages_array(
+        {1, 2, 3},
+        {4},
+        {1,2}
+    )
+    assert resulting_set == {3, 4}
+
+
+def test_kafka_message_needed(annotated_doc: AnnotatedDoc):
+    db = Mock()
+    check_if_kafka_message_is_needed(
+        db,
+        annotated_doc,
+        annotated_doc,
+        annotated_doc.job_id,
+        annotated_doc.file_id,
+        TEST_TENANT
+    )
+    
+    
+def test_kafka_message_needed_commit(annotated_doc: AnnotatedDoc):
+    db = Mock()
+    db.commit = Mock()
+    check_if_kafka_message_is_needed(
+        db,
+        annotated_doc,
+        None,
+        annotated_doc.job_id,
+        annotated_doc.file_id,
+        TEST_TENANT
+    )
+    db.commit.assert_called_once()
```

Comment:
Add new line here.

Reply:
It looks like your pre-commit hook isn't working; otherwise, it wouldn't have allowed you to commit such code.